q2_amr/types/_format.py

```diff
@@ -430,3 +430,215 @@ class CARDGeneAnnotationDirectoryFormat(MultiDirValidationMixin, model.Directory
+            "Single species (plasmid) reads",
+            "Single species (no genomic info) reads",
+            "Single genus (chromosome) reads",
+            "Single genus (chromosome or plasmid) reads",
+            "Single genus (plasmid) reads",
+            "Single genus (no genomic info) reads",
+            "Promiscuous plasmid reads",
+            "Unknown taxonomy (chromosome) reads",
+            "Unknown taxonomy (chromosome or plasmid) reads",
+            "Unknown taxonomy (no genomic info) reads",
+        ]
+
+        df = pd.read_csv(str(self), sep="\t")
+        header_obs = list(df.columns)
+        if not set(header_exp).issubset(set(header_obs)):
+            raise ValidationError(
+                "Header line does not match CARDReadsGeneKmerAnalysisFormat. Must "
+                "contain the following values: "
+                + ", ".join(header_exp)
+                + ".\n\nFound instead: "
+                + ", ".join(header_obs)
+            )
+
+    def _validate_(self, level):
+        self._validate()
+
+
+class CARDReadsAlleleKmerAnalysisFormat(model.TextFileFormat):
+    def _validate(self, n_records=None):
+        header_exp = [
+            "Reference Sequence",
+            "Mapped reads with kmer DB hits",
+            "CARD*kmer Prediction",
+            "Single species (chromosome) reads",
+            "Single species (chromosome or plasmid) reads",
+            "Single species (plasmid) reads",
+            "Single species (no genomic info) reads",
+            "Single genus (chromosome) reads",
+            "Single genus (chromosome or plasmid) reads",
+            "Single genus (plasmid) reads",
+            "Single genus (no genomic info) reads",
+            "Promiscuous plasmid reads",
+            "Unknown taxonomy (chromosome) reads",
+            "Unknown taxonomy (chromosome or plasmid) reads",
+            "Unknown taxonomy (no genomic info) reads",
+        ]
+
+        df = pd.read_csv(str(self), sep="\t")
+        header_obs = list(df.columns)
+        if not set(header_exp).issubset(set(header_obs)):
+            raise ValidationError(
+                "Header line does not match CARDReadsAlleleKmerAnalysisFormat. Must "
+                "contain the following values: "
+                + ", ".join(header_exp)
+                + ".\n\nFound instead: "
+                + ", ".join(header_obs)
+            )
+
+    def _validate_(self, level):
+        self._validate()
+
+
+class CARDReadsKmerAnalysisJSONFormat(model.TextFileFormat):
+    def _validate(self, n_records=None):
+        keys_exp = [
+            "reference",
+            "#_of_kmers_in_sequence",
+            "#_of_AMR_kmers",
+            "SAM_flag",
+            "MAPQ",
+            "taxonomic_info",
+            "genomic_info",
+        ]
+        with open(str(self)) as json_file:
+            dict = json.load(json_file)
+
+        keys_obs = list(next(iter(dict.values())).keys())
+
+        if keys_obs != keys_exp:
```

Comment:
Same as my previous comment, change to comparing sets if you don't care about enforcing order to the columns.